### ISSUE TITLE
General module for automatically toggling NVDA on/off

### ISSUE THREAD

**Issue by nvdaTrac** [nvaccess/test_nvdaTracImport#264]
**Reported by tvalusek on 27 Sep 2008 15:39 UTC**
**Blocking #183**
an approach introduced by DosVox.py should be generalized into a new preferences category with apps where NVDA is not needed, either partially or completely. Besides Dosvox, there are other apps which maintain interaction with a blind user by themselves, such as czech software suite Elvos Asistent, or Microsoft reader, where it is possible to turn on extended speech feedback.

**Comment by nvdaTrac** [comment]
**Modified by jteh on 5 Dec 2008 03:23 UTC**

**Comment by nvdaTrac** [comment]
**Comment by jteh on 16 Jan 2009 01:26 UTC**
I have received another request for a sleep mode feature.

 * I'm not convinced we need a new preferences style system to handle this. Instead, it could be done in appModules, but perhaps with a class that could simply be subclassed in each appModule. For example:
```
import appModuleHandler
class AppModule(appModuleHandler.SleepAppModule):
	pass
```
 * Should we suspend handling of events as well or just keyboard input, speech and braille?
  * Suspending events is trickier.
  * We can't suspend all events anyway, as we still need to handle gainFocus events to know when to stop sleeping.

**Comment by nvdaTrac** [comment]
**Comment by jteh on 18 May 2009 07:31 UTC**
Implemented in r2852.

**Comment by nvdaTrac** [comment]
**Comment by jteh on 18 May 2009 07:32 UTC**
We'll try closing it this time...
